nucypher/blockchain/eth/deployers.py

```diff
@@ -151,26 +154,149 @@ def _ensure_contract_deployment(self) -> bool:
             raise self.ContractDeploymentError(message)
         return True
 
-    def deploy(self, secret_hash: bytes, gas_limit: int, progress) -> dict:
+    def deploy(self, gas_limit: int, progress) -> dict:
         """
         Provides for the setup, deployment, and initialization of ethereum smart contracts.
         Emits the configured blockchain network transactions for single contract instance publication.
         """
-        # TODO: Consider splitting this into to classes upgradeable v non-upgraeable.
-        #       Alternately, just remove deployment secrets alltogether.
         raise NotImplementedError
 
     def make_agent(self) -> EthereumContractAgent:
         agent = self.agency(registry=self.registry, contract=self._contract)
         return agent
 
 
-class NucypherTokenDeployer(ContractDeployer):
+
+class OwnableContractMixin:
+
+    _ownable = True
+
+    class NotOwnable(RuntimeError):
```

Comment:
This sounds odd to me - to raise `NotOwnable`. Would it be better to say `ContractNotOwnable`?

Reply:
:Yeah - Good Idea. :heavy_check_mark: